ldap/servers/plugins/automember/automember.c

```diff
@@ -2285,9 +2351,22 @@ automember_task_add(Slapi_PBlock *pb, Slapi_Entry *e, Slapi_Entry *eAfter __attr
     const char *base_dn;
     const char *filter;
     const char *scope;
+    const char *cleanup_str;
+    PRBool cleanup = PR_FALSE;
 
     *returncode = LDAP_SUCCESS;
 
+    PR_Lock(fixup_lock);
+    if (fixup_running) {
+        PR_Unlock(fixup_lock);
+        *returncode = LDAP_UNWILLING_TO_PERFORM;
+        slapi_log_err(SLAPI_LOG_ERR, AUTOMEMBER_PLUGIN_SUBSYSTEM,
+                "automember_task_add - there is already a fixup task running\n");
```

Comment:
Could you please add a small test (or just a check to the existing test) - for this addition?
Something like - trying to add a few tasks at the same time, and checking if the error is displayed.

Reply:
Done